packages/expect/src/asymmetricMatchers.ts

```diff
@@ -257,8 +257,8 @@ class StringContaining extends AsymmetricMatcher<string> {
     super(sample, inverse);
   }
 
-  asymmetricMatch(other: string) {
-    const result = isA('String', other) && other.includes(this.sample);
+  asymmetricMatch(other: unknown) {
+    const result = isA<string>('String', other) && other.includes(this.sample);
```

Comment:
This is just an internal type, user facing external types do not change. The point is that internally `other` is of type `unknown` in all the matchers. As an illustration see this test (all matchers have similar tests and that is a problem for typechecking):

https://github.com/facebook/jest/blob/8759d63787b832af1fc9ac0ec2cc57b6f325ba9a/packages/expect/src/__tests__/asymmetricMatchers.test.ts#L321

At first it sounded strange, but reasons are very well explained in https://github.com/facebook/jest/pull/7107